demo/src/utils.js

```diff
@@ -28,36 +28,41 @@ export const getKeyringPair = async (ss58Format) => {
   return keyringPair;
 };
 
-export const sendExtrinsic = (extrinsic, api, keyringPair) =>
-  new Promise((resolve, reject) => {
-    const signAndSend = async () => {
-      const unsub = await extrinsic.signAndSend(keyringPair, (result) => {
-        const { status, events, dispatchError } = result;
-        console.log("status.type: ", status.type);
-
-        if (status?.isFinalized) {
-          unsub();
-          if (!_.isNil(dispatchError)) {
-            reject(dispatchError);
-          }
-
-          const event = _.find(events, ({ event: eventData }) =>
-            api.events.system.ExtrinsicSuccess.is(eventData),
-          );
-          if (event) {
-            resolve({
-              blockHash: status?.asFinalized?.toString(),
-              events,
-              extrinsicHash: extrinsic.hash,
-            });
-          } else {
-            reject(new Error("The event.ExtrinsicSuccess is not found"));
-          }
-        }
-      });
-    };
-    signAndSend();
-  });
+export const sendExtrinsic = (api, extrinsic, keyPair) =>
+	new Promise((resolve, reject) => {
+		const signAndSend = async () => {
+			try {
+				const unsub = await extrinsic.signAndSend(keyPair, (result) => {
+					const { status, events, dispatchError } = result;
+					console.log("status.type: ", status.type);
+
+					if (status?.isFinalized) {
+						unsub();
+
+						if (dispatchError) {
+							return reject(new Error(dispatchError));
+						}
+
+						const event = _.find(events, ({ event: eventData }) => api.events.system.ExtrinsicSuccess.is(eventData));
+						if (event) {
+							return resolve({
+								blockHash: status?.asFinalized?.toString(),
+								events,
+								extrinsicHash: extrinsic?.hash?.toString(),
+							});
+						} else {
+							return reject(new Error(events.toString()));
+						}
+					}
+				});
+			} catch (ex) {
+				// Handle signing error such as user manually cancel the transaction
+				return reject(ex);
+			}
```

Comment:
Without this catch and reject in it, this function cannot return immediate exceptions (for example, something is wrong during signing)